src/tools/synthetic_node.rs

```diff
@@ -119,13 +138,14 @@ impl SyntheticNodeBuilder {
     }
 }
 
-/// Conventient abstraction over the `pea2pea`-backed node to be used in tests.
+/// Conventient abstraction over a `pea2pea` node.
 pub struct SyntheticNode {
     inner_node: InnerNode,
     inbound_rx: Receiver<(SocketAddr, Message)>,
 }
 
 impl SyntheticNode {
+    // FIXME: remove in favour of calling `SyntheticNodeBuilder::default()` or `new` directly?
```

Comment:
I like this way around since it makes explicit that this is a builder for `SyntheticNode`. Maybe even removing the `Default` for the builder and make it a private `new()` - so the only way of constructing one is via `SyntheticNode::builder()`?

It's not a strong opinion though, so happy for it go either way.